pkg/virt-controller/watch/preset.go

```diff
@@ -152,6 +152,9 @@ func (c *VirtualMachinePresetController) initializeVirtualMachine(vm *kubev1.Vir
 		if !success {
 			logger.Object(vm).Warning("Marking VM as failed")
 			vm.Status.Phase = kubev1.Failed
+		} else {
+			logger.Object(vm).V(4).Info("Setting default values on VirtualMachine")
+			kubev1.SetObjectDefaults_VirtualMachine(vm)
```

Comment:
We need to be careful to not override preset changes with our default values, maybe does it better to apply default values before preset values?

Reply:
defaults are only set when sth is missing, it won't override anything. So I think first setting presets, and then defaults, is the correct order. Agree?

Reply:
If we assume that `SetObjectDefaults_VirtualMachine` does not have bugs, I agree :smiley: :

Reply:
Yes, presets first. We also want to ensure that structs set or added by presets get filled with missing defaults.